### PR TITLE
find-replace blog post

### PR DESCRIPTION
**Pull request by beyang**
Educational blog post surveying find-replace tools. Features Comby and Campaigns.

### CODE REVIEW

File: blogposts/2020/find-and-replace.md [hunk 1]
```diff
@@ -0,0 +1,582 @@
...
+or even years to fully propagate across the entire codebase of an organization. This is a big enough
+problem that many dev organizations have created tools to facilitate large-scale changes to code
+that span many files, teams, and repositories. These tools and the associated processes go by
+various names such as "large-scale refactoring", "large-scale codemods", and "code
+shepherding". These tools are usually specific to the organization that created them and rarely
+released publicly.
+
+### Sourcegraph Campaigns
+
+After hearing about many similar challenges from many of our customers, our team at Sourcegraph
```
Comment (sqs) [inline blogposts/2020/find-and-replace.md:454]: This is product/business-speak, not developer-speak :P

File: blogposts/2020/find-and-replace.md [hunk 2]
```diff
@@ -0,0 +1,582 @@
...
+
+We're calling this feature Sourcegraph Campaigns. If Sourcegraph Code Search is the "find" operation
+over all your code, then Sourcegraph Campaigns is the "replace".
+
+Suppose your organization has decided it wants to standardize error handling across your
+codebase. In particular, you want to use [wrapped errors](https://blog.golang.org/go1.13-errors) in
+Go code. This means you want to ensure that instances of `fmt.Errorf` use the `%w` format verb,
+rather than `%s`.
+
+Without Campaigns, this would involve cloning down all your repositories one-by-one, running a
```
Comment (sqs) [inline blogposts/2020/find-and-replace.md:466]: lowercase campaigns, don't refer to things as "Sourcegraph xyz" above. It's ok and actually helpful to do that in marketing copy, but not in this kind of post IMO.

File: blogposts/2020/find-and-replace.md [hunk 3]
```diff
@@ -0,0 +1,582 @@
...
+* You're identifying and fixing common anti-patterns to improve overall code quality.
+
+Doing any of these by hand can be mind-numbing. For assistance, you can turn to a wide variety of
+find-and-replace tools. But there are so many such tools, with varying degrees of applicability,
+expressivity, ease-of-use, ease-of-learning, and scalability. How do you pick the right one?
+
+This post presents a guide to all the various find-and-replace tools I've encountered and used over
+the years, divided into three chapters:
+
+1. [Find and replace in your editor](#find-and-replace-in-your-editor)
```
Comment (sqs) [inline blogposts/2020/find-and-replace.md:34]: Could you represent the available tools in a matrix to show that there's a gap where campaigns fits in?

Not sure what the rows and columns would be

File: blogposts/2020/find-and-replace.md [hunk 4]
```diff
@@ -0,0 +1,582 @@
...
+|-----------------------------------------------------|------------|--------------------|----------------|
+| Find all symbol names starting with "http"          | `foo\w*`   | fooBar             | barFoo         |
+| Find all characters between double quotes           | `"[^"]"`   | "hello&nbsp;world" | hello world'   |
+| Find all references to fields of a certain variable | `\w+\.\w+` | base.Path          | basePath       |
+
+There are different dialects or flavors of regex. In this post, we'll use [**POSIX Extended Regular
+Expressions**](https://www.gnu.org/software/findutils/manual/html_node/find_html/posix_002dextended-regular-expression-syntax.html#posix_002dextended-regular-expression-syntax). Here's
+a short primer on its syntax:
+
+* Alphanumeric characters are generally interpreted literally.
```
Comment (sqs) [inline blogposts/2020/find-and-replace.md:81]: Feels too elementary/reference-like to be showing this here. Who's the audience? What's the goal?

I think the audience should be senior engineers and the goal should be to make people realize "hmm, yeah, i as a dev have made the progression from find in my editor, to grep, to codemod...and there is a gap in my needs that something like campaigns fills (or the next step in my evolution is something like campaigns)...oh, and google's been using something like it for a while too?" (and check out what fatih mentions at https://twitter.com/fatih/status/1259912881186824192 and cite that too)

File: blogposts/2020/find-and-replace.md (hunk 3 above)
Comment (beyang) [reply blogposts/2020/find-and-replace.md:95]: Added this: 
Selection_161

**Pull request merged**